File: SantaFinder.Web/Models/ChangeOrder/OrderFullInfoForEditing.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using SantaFinder.Entities;
using SantaFinder.Web.Models.OrderHistory;
using SantaFinder.Web.Models.Santas;
using SantaFinder.Web.Models.Shared;

namespace SantaFinder.Web.Models.ChangeOrder
{
    public class OrderFullInfoForEditing
    {
        public int Id { get; set; }
        public string ChildrenNames { get; set; }
        public DateTime Datetime { get; set; }
        public Address Address { get; set; }
        public Location Location { get; set; }
        public OrderStatus Status { get; set; }
        public IEnumerable<PresentInfoForEdit> Presents { get; set; }
        public SantaInfoForClient SantaInfo { get; set; }
        public int? Rating { get; set; }
    }
}

File: SantaFinder.Web/Client/src/app/client/order-info/order-full-info-for-editing.ts
import { Address, Santa, ChatMessage } from '../../core/models';
import { OrderStatus } from '../../core/enums';
import { PresentInfoForEditing } from './present-info-for-editing';

export interface OrderFullInfoForEditing {
    id: number;
    childrenNames: string;
    datetime: Date;
    address: Address;
    location: Location;
    status: OrderStatus;
    presents: PresentInfoForEditing[];
    santaInfo: Santa;
    rating: number;
    chatMessages: ChatMessage[];
}